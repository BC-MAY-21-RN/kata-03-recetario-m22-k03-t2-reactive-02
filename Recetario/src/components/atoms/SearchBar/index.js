import React from 'react';
import {SearchBar} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';

const icons = <Icon name="search" size={18} color="#ffffff" />;

const Search = props => {
  return (
    <SearchBar
      placeholder={props.placeholder}
      onChangeText={props.onChangeText}
      value={props.value}
      searchIcon={icons}
    />
  );
};

export default Search;
