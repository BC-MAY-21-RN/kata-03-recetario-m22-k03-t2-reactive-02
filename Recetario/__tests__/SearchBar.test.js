import 'react-native';
import React from 'react';
import SearchBar from '../src/components/atoms/SearchBar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
