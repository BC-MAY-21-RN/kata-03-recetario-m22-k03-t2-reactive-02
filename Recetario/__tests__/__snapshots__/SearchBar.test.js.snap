// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#282828",
      "borderBottomColor": "transparent",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#343435",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 10,
          "flexDirection": "row",
          "margin": 5,
          "minHeight": 30,
          "overflow": "hidden",
          "padding": 5,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï€‚
        </Text>
      </View>
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#86939e"
        platform="default"
        style={
          Object {
            "color": "#86939e",
            "flex": 1,
            "fontSize": 18,
            "marginLeft": 10,
            "minHeight": 40,
          }
        }
        testID="searchInput"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginRight": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
</View>
`;
